pkg/controller/metrics/metrics.go

```diff
@@ -127,6 +156,9 @@ var (
 		},
 		[]string{"cluster_deployment", "namespace", "cluster_type"},
 	)
+
+	// MapMetricToDuration is a map of optional durationMetrics to their specific duration, if mentioned
+	MapMetricToDuration map[*prometheus.HistogramVec]time.Duration
```

Comment:
This var is only used in this package, so let's make it private.

(We could also make it an object, with a factory method to hide the fact that it's a package global, and `registerOptionalMetrics` and `ShouldLogDurationMetric` as methods on it. But le big meh.)